changes required to allow replacing the obsolete __conda_version__ in conda-rdkit

ptosco opened the pull request:
These changes are required to allow replacing the obsolete __conda_version__ in conda-rdkit, and get the rdkit and rdkit-postgresql conda build recipes to work again (see issue #53).

greglandrum commented on Code/cmake/Modules/RDKitVersion.cmake:
```diff
@@ -0,0 +1,21 @@
+set(RDKit_Year "2018")
```
Changing these values in this file seems odd (and a bit opaque) to me.
Will this patch work if we leave the definitions of these values in the root `CMakeLists.txt` file?

ptosco replied:
Yes,I think it should still work. I’ll check it later.

greglandrum commented on pkg_version.py:
```diff
@@ -0,0 +1,67 @@
+import os
```
I'd like this file to be somewhere other than the root of the code base. Since it's just being used by the conda stuff (at least I assume so), maybe it could go in the rkdit-conda repo?

ptosco replied:
I think I have an idea how to make it work. Again, I’ll check it later.

rvianello replied:
I was looking at the load_setup_py_data details in conda build, and it looks like it might accept a path to a recipe directory, where this script could be located (https://github.com/conda/conda-build/blob/master/conda_build/jinja_context.py#L82).

ptosco commented:
Unfortunately `pkg_version.py` cannot be moved to the `conda-rdkit` repo. The reason is that when `load_setup_py_data()` sources the script, none of the `conda` environment variables (i.e., `SRC_DIR`, `RECIPE_DIR`, etc.) are passed to the `pkg_version.py` script. So the only version for the script to know where `SRC_DIR` is in order to find and parse `CMakeLists.txt` and `RDKitUtils.cmake` is to be located in the `SRC_DIR` tree, so it can retrieve the `SRC_DIR` path.
To make things tidier I have moved `pkg_version.py`  to a `Conda` subdirectory - this is the best I can do I am afraid.
I have also reverted all other changes (i.e., `CMakeLists.txt` and `RDKitUtils.cmake`). So the only change in the RDKit trunk is the addition of `Conda/pkg_version.py` - I hope this is acceptable.

greglandrum commented:
One minor request: would you mind renaming the directory to`build_support`? It's intentional that this is somewhat inconsistent with the other directory names.

ptosco commented:
Done.

greglandrum commented:
Thanks Paolo
